webpack/assets/javascripts/react_app/components/AutoComplete/index.js

```diff
@@ -23,18 +23,42 @@ class AutoComplete extends React.Component {
       'handleInputFocus',
       'getResults',
       'windowKeyPressHandler',
-      'unableHTMLAutocomplete',
+      'disableHTMLAutocomplete',
       'handleKeyDown',
     ]);
     this._typeahead = React.createRef();
     debounceMethods(this, 500, ['handleLoading']);
+    // Used to pass _typeahead into the static function getDerivedStateFromProps.
+    this.state = {
+      _typeahead: this._typeahead,
+    };
+  }
+
+  static getDerivedStateFromProps({ trigger }, { _typeahead }) {
```

Comment:
`getDerivedStateFromProps` is invoked on each and every render.  and it meant for change the state from props, andd should be used sparingly. here it's not the case.

from react docs:
> "getDerivedStateFromProps exists for only one purpose. It enables a component to update its internal state as the result of changes in props.... "

Reply:
as it replaced the `componentWillRecieveProps`
I do think it is right to use in this case, and it doesn't must update the state.

from react docs: 

> getDerivedStateFromProps is invoked right before calling the render method, both on the initial mount and on subsequent updates. It should return an object to update the state, or null to update nothing.

I am open to implement this functionality with other lifecycle method if you have any better idea.

Reply:
https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html

Reply:
Sounds good, I will try that !

Reply:
Switched to `componentDidUpdate`